wave_lang/kernel/wave/runtime/runtime.cpp

```diff
@@ -193,21 +194,31 @@ static void launch(const KernelLaunchInfo &info, const Int64Vector &tensors,
   }
 }
 
-static void unload_binary(void *ptr) noexcept {
-  auto module = reinterpret_cast<hipModule_t>(ptr);
-  if (auto e = hipModuleUnload(module)) {
-    nb::print(nb::str("Failed to unload module: ") +
-              nb::str(hipGetErrorString(e)));
+static std::vector<char> readFileIntoVector(const std::string &filename) {
+  std::ifstream file(filename, std::ios::binary | std::ios::ate);
+
+  if (!file.is_open()) {
+    std::cerr << "Unable to open file: " << filename << std::endl;
+    return std::vector<char>();
   }
+
+  std::streamsize size = file.tellg();
```

Comment:
https://stackoverflow.com/questions/22984956/tellg-function-give-wrong-size-of-file/22986486#22986486

Reply:
If you have a somewhat modern C++ standard library, it should have native support https://en.cppreference.com/w/cpp/filesystem/file_size.html.

Reply:
change it to newer API